tcms/core/tests/test_forms.py

```diff
@@ -0,0 +1,67 @@
+# -*- coding: utf-8 -*-
+# pylint: disable=invalid-name
+from django.test import TestCase
+from django.core.exceptions import ValidationError
+
+from tcms.core.forms.fields import UserField
+from tcms.tests.factories import UserFactory
+
+
+class TestUserField(TestCase):
+
+    @classmethod
+    def setUpTestData(cls):
+        cls.user = UserFactory(username='admin', email='admin@example.com')
+        cls.user.set_password('admin')  # nosec:B106:hardcoded_password_funcarg
+        cls.user.is_superuser = True
+        cls.user.is_staff = True  # enables access to admin panel
+        cls.user.save()
+
+        cls.user_field = UserField()
+
+    def test_valid_username_accepted(self):
+        form_user = self.user_field.clean("admin")
+        self.assertEqual(form_user.email, self.user.email)
+        self.assertEqual(form_user.username, self.user.username)
+
+    def test_empty_username_when_required_throws_error(self):
+        with self.assertRaisesRegex(ValidationError, 'A user name or user ID is required.'):
+            self.user_field.clean('')
+
+    def test_none_username_when_required_throws_error(self):
+        with self.assertRaisesRegex(ValidationError, 'A user name or user ID is required.'):
+            self.user_field.clean(None)
+
+    def test_none_username_accepted_when_not_required(self):
+        self.user_field.required = False
+        self.assertIsNone(self.user_field.clean(None))
+        self.user_field.required = True
+
+    def test_empty_username_accepted_when_not_required(self):
+        self.user_field.required = False
+        self.assertIsNone(self.user_field.clean(''))
+        self.user_field.required = True
+
+    def test_int_instance_username_accepted(self):
+        form_user = self.user_field.clean(int(self.user.pk))
+        self.assertEqual(form_user.email, self.user.email)
+        self.assertEqual(form_user.username, self.user.username)
+
+    def test_not_existing_int_instance_username_throws_error(self):
+        user_id = int(self.user.pk + 1)
+        with self.assertRaisesRegex(ValidationError, 'Unknown user_id: \"' + str(user_id) + '\"'):
```

Comment:
Type casting should be avoided, self.user.pk is a <class 'int'> already. Also avoid concatenation of a string, instead use string formatting something like  "Unknown user_id: %d" % user_id.

Reply:
Removed the re-casting to `int ` with latest commit.